src/rgw/rgw_log.cc

```diff
@@ -128,14 +128,13 @@ class UsageLogger {
     round_timestamp = ts.round_to_hour();
   }
 
-  void insert_user(utime_t& timestamp, const rgw_user& user, rgw_usage_log_entry& entry) {
+  void insert_user(utime_t& timestamp, const string& user, rgw_usage_log_entry& entry) {
```

Comment:
Anyway, even if we had introduced such restriction, it wouldn't help with already deployed clusters. :-( Keeping the backward compatibility is a must.

We would rather convey a tuple consisted of `tenant`, `user` and `subuser` across the usage logging code. Also, we would need to still account aggregated statistics at per-user level. This most likely means extending `rgw_usage_log_entry` and `rgw_user_bucket` to convey also the (optional) `subuser` part. Those links might be useful:
* https://github.com/ceph/ceph/blob/master/src/rgw/rgw_rados.h#L248,
* https://github.com/ceph/ceph/blob/master/src/cls/rgw/cls_rgw.cc#L2787.